src/string/drain.rs

```diff
@@ -8,40 +10,40 @@ use super::StringView;
 /// documentation for more.
 ///
 /// [`drain`]: crate::String::drain
-pub struct Drain<'a> {
+pub struct Drain<'a, LenT: LenType> {
     /// Will be used as &'a mut String in the destructor
-    pub(super) string: *mut StringView,
+    pub(super) string: *mut StringView<LenT>,
     /// Stast of part to remove
-    pub(super) start: usize,
+    pub(super) start: LenT,
     /// End of part to remove
-    pub(super) end: usize,
+    pub(super) end: LenT,
     /// Current remaining range to remove
     pub(super) iter: Chars<'a>,
 }
 
-impl fmt::Debug for Drain<'_> {
+impl<LenT: LenType> fmt::Debug for Drain<'_, LenT> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         f.debug_tuple("Drain").field(&self.as_str()).finish()
     }
 }
 
-unsafe impl Sync for Drain<'_> {}
-unsafe impl Send for Drain<'_> {}
+unsafe impl<LenT: LenType> Sync for Drain<'_, LenT> {}
+unsafe impl<LenT: LenType> Send for Drain<'_, LenT> {}
 
-impl Drop for Drain<'_> {
+impl<LenT: LenType> Drop for Drain<'_, LenT> {
     fn drop(&mut self) {
         unsafe {
             // Use `Vec::drain`. “Reaffirm” the bounds checks to avoid
             // panic code being inserted again.
             let self_vec = (*self.string).as_mut_vec();
-            if self.start <= self.end && self.end <= self_vec.len() {
-                self_vec.drain(self.start..self.end);
+            if self.start <= self.end && self.end <= self_vec.length() {
+                self_vec.drain(self.start.into_usize()..self.end.into_usize());
```

Comment:
Convert them before the `if`, so you don't need the `length` method.

Reply:
Done.